chaco/tools/legend_highlighter.py

```diff
@@ -54,35 +57,34 @@ def normal_left_down(self, event):
             return
 
         plots = get_hit_plots(self.component, event)
-
-        if len(plots) > 0:
-            plot = plots[0]
-
-            if event.shift_down:
-                # User in multi-select mode by using [shift] key.
+        if event.shift_down:
+            # User in multi-select mode by using [shift] key.
+            for plot in plots:
                 if plot in self._selected_renderers:
                     self._selected_renderers.remove(plot)
                 else:
                     self._selected_renderers.append(plot)
-
-            else:
-                # User in single-select mode.
-                add_plot = plot not in self._selected_renderers
-                self._selected_renderers = []
-                if add_plot:
-                    self._selected_renderers.append(plot)
-
-            if self._selected_renderers:
-                self._set_states(self.component.plots)
-            else:
-                self._reset_selects(self.component.plots)
-            plot.request_redraw()
+        elif plots:
+            # User in single-select mode.
+            add_plot = any(plot not in self._selected_renderers
+                           for plot in plots)
+            self._selected_renderers = []
+            if add_plot:
+                self._selected_renderers.extend(plots)
+
+        if self._selected_renderers:
+            self._set_states(self.component.plots)
+        else:
+            self._reset_selects(self.component.plots)
+
+        if plots:
+            plots[0].request_redraw()
 
         event.handled = True
 
     def _reset_selects(self, plots):
         """ Set all renderers to their default values. """
-        for plot in sm.reduce(operator.add, plots.values()):
+        for plot in sum((_ensure_list(p) for p in plots.values()), []):
```

Comment:
`plots` is the dictionary of the same name on the `Legend` instance. We need to anticipate `values()` being single renderers or lists of renderers.

Reply:
A better construct for concatenating list-likes is `itertools.chain.from_iterable` to the point where I sometimes alias that as `concat`.

Reply:
Yup. Done.